tests/Unit/Parallel/Actions/Test_SetupDataBox.cpp

```diff
@@ -0,0 +1,96 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <string>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Framework/ActionTesting.hpp"
+#include "Parallel/Actions/SetupDataBox.hpp"  // IWYU pragma: keep
+#include "Parallel/Actions/TerminatePhase.hpp"
+#include "Parallel/PhaseDependentActionList.hpp"  // IWYU pragma: keep
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+struct Label1;
+struct Label2;
+
+struct CounterTag : db::SimpleTag {
+  using type = size_t;
+};
+
+struct VectorTag : db::SimpleTag {
+  using type = DataVector;
+};
+
+struct SquareVectorTag : db::SimpleTag {
+  using type = DataVector;
+};
+
+struct SquareVectorCompute : SquareVectorTag, db::ComputeTag {
+  using argument_tags = tmpl::list<VectorTag>;
+  static void function(const gsl::not_null<DataVector*> result,
+                       const DataVector& vector) noexcept {
+    *result = square(vector);
+  }
+  using return_type = DataVector;
+  using base = SquareVectorTag;
+};
+
+
+struct InitializationAction {
```

Comment:
How about you add this example to the documentation of `SetupDataBox` as a `\snippet`?